azurerm/resource_arm_cosmosdb_mongo_collection.go

```diff
@@ -0,0 +1,301 @@
+				ValidateFunc: validate.CosmosEntityName,
+			},
+
+			// SDK/api accepts an array.. but only one is allowed
+			"shard_key": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.NoEmptyStrings,
+			},
+
+			// default TTL is simply an index on _ts with expireAfterOption, given we can't seem to set TTLs on a given index lets expose this to match the portal
+			"default_ttl_seconds": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ValidateFunc: validation.IntAtLeast(-1),
+			},
+
+			"indexes": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"key": {
+							Type:         schema.TypeString, // this is a list in the SDK/API, however any more then a single value causes a 404
+							Required:     true,
+							ValidateFunc: validate.NoEmptyStrings,
+						},
+
+						"unique": {
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  true, // portal defaults to true
+						},
+
+						// expire_after_seconds is only allowed on `_ts`:
+						// Unable to parse request payload due to the following reason: 'The 'expireAfterSeconds' option is supported on '_ts' field only.
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceArmCosmosDbMongoCollectionCreateUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).cosmosAccountsClient
+	ctx := meta.(*ArmClient).StopContext
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+	account := d.Get("account_name").(string)
+	database := d.Get("database_name").(string)
+
+	if requireResourcesToBeImported && d.IsNewResource() {
+		existing, err := client.GetMongoDBCollection(ctx, resourceGroup, account, database, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("Error checking for presence of creating Cosmos Mongo Collection %s (Account %s, Database %s): %+v", name, account, database, err)
+			}
+		} else {
+			id, err := azure.CosmosGetIDFromResponse(existing.Response)
+			if err != nil {
+				return fmt.Errorf("Error generating import ID for Cosmos Mongo Collection %s (Account %s, Database %s)", name, account, database)
+			}
+
+			return tf.ImportAsExistsError("azurerm_cosmosdb_mongo_collection", id)
+		}
+	}
+
+	var ttl *int
+	if v, ok := d.GetOkExists("default_ttl_seconds"); ok {
+		ttl = utils.Int(v.(int))
+	}
+
+	db := documentdb.MongoDBCollectionCreateUpdateParameters{
+		MongoDBCollectionCreateUpdateProperties: &documentdb.MongoDBCollectionCreateUpdateProperties{
+			Resource: &documentdb.MongoDBCollectionResource{
+				ID:      &name,
+				Indexes: expandCosmosMongoCollectionIndexes(d.Get("indexes"), ttl),
+			},
+			Options: map[string]*string{},
+		},
+	}
+
+	if v, ok := d.GetOkExists("shard_key"); ok {
+		db.MongoDBCollectionCreateUpdateProperties.Resource.ShardKey = map[string]*string{ // errors
```

Comment:
Is `// errors` needed?
```suggestion
		db.MongoDBCollectionCreateUpdateProperties.Resource.ShardKey = map[string]*string{ 
```